.github/workflows/publish.yml

```diff
@@ -0,0 +1,53 @@
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies and build 🔧
+        run: npm ci && npm run build
+      - name: Run lint 
+        run: npm run lint
+      - name: Run tests 
+        run: npm run test
+      - name: Package module
+        run: npm pack
+      - name: Determine version
+        id: version
+        run: echo "version=${GITHUB_REF:11}" >> $GITHUB_OUTPUT
```

Comment:
I've noticed that we use different logic to determine the package version here and in ci.yml.

Do we need to check that the tag name matches package.json version name, or does `npm publish` take care of that? (What happens if we tag a new version but forget to bump the version number in package.json?)

Reply:
The difference is that in any regular PR the version is not part of the tag here. We are getting the version from package.json because `npm pack` will generate a `esptool-js-<version>.tgz` which we use to create a PR artifact that could be used for review/testing. 

In the publish.yml workflow, we use the GITHUB_REF because the workflow is triggered on tags with `v*` to do the release the module in npm. While npm publish doesn't bump the version in package.json, but if the tag already exist in the GitHub repository this workflow would not run (i.e if version already exists) so new version is required for release.

We could modify it so this step gets the version from package.json, but the triggering mechanism remains as described before.

Reply:
The only case I'm worried about is if we tag a new version but forget to update the version number in package.json.

To me it seems like an additional check in the workflow for tags would solve this. That is, extract the version from package.json and verify if it's the same as the tag name.